addons/mail/static/src/discuss_app/discuss.js

```diff
@@ -100,13 +106,24 @@ export class Discuss extends Component {
         return this.store.threads[this.store.discuss.threadLocalId];
     }
 
+    get channelAvatar() {
+        return this.props.public
+            ? url(`/discuss/channel/${this.thread.id}/avatar_128`)
+            : this.thread.imgUrl;
+    }
+
     togglePinMenu() {
         this.state.activeMode =
             this.state.activeMode === this.MODES.PINNED_MESSAGES
                 ? this.MODES.NONE
                 : this.MODES.PINNED_MESSAGES;
     }
 
+    async onFileUploaded(file) {
+        await this.threadService.updateAvatar(this.thread.id, file.data);
```

Comment:
crash: `this.threadService.updateAvatar is not a function` (tested on public page as admin)